From 337a6eabbefd2cfb8176c51dd481bf13202f8dd2 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:40:15 -0400
Subject: [virt] kvm/async_pf: change async_pf_execute() to use get_user_pages(tsk => NULL)

Message-id: <1406040016-3289-221-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86299
O-Subject: [RHEL7 PATCH v2 220/221] KVM: async_pf: change async_pf_execute() to use get_user_pages(tsk => NULL)
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Oleg Nesterov <oleg@redhat.com>

async_pf_execute() passes tsk == current to gup(), this is doesn't
hurt but unnecessary and misleading. "tsk" is only used to account
the number of faults and current is the random workqueue thread.

Signed-off-by: Oleg Nesterov <oleg@redhat.com>
Suggested-by: Andrea Arcangeli <aarcange@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit e9545b9f8aeb63e05818e4b3250057260bc072aa)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/virt/kvm/async_pf.c b/virt/kvm/async_pf.c
index f2c80d5..7580aa1 100644
--- a/virt/kvm/async_pf.c
+++ b/virt/kvm/async_pf.c
@@ -67,7 +67,7 @@ static void async_pf_execute(struct work_struct *work)
 
  use_mm(mm);
  down_read(&mm->mmap_sem);
- get_user_pages(current, mm, addr, 1, 1, 0, NULL, NULL);
+ get_user_pages(NULL, mm, addr, 1, 1, 0, NULL, NULL);
  up_read(&mm->mmap_sem);
  unuse_mm(mm);
 
-- 
1.7.1